libclamav/readdb.c

```diff
@@ -3411,25 +3412,13 @@ static int cli_loadcrt(FILE *fs, struct cl_engine *engine, struct cli_dbio *dbio
             goto done;
         }
 
-        pubkey = cli_hex2str(tokens[4]);
-        if (!pubkey) {
+        if (BN_hex2bn(&ca.n, tokens[4]) == 0) {
             cli_errmsg("cli_loadcrt: line %u: Cannot convert public key to binary string\n", (unsigned int)line);
             ret = CL_EMALFDB;
             goto done;
         }
-
-        /*
-         * tokens[4] is the public key.  having a length that is too
-         * long causes an out of bounds read in the this call.
-         */
-        if ((strlen(tokens[4]) / 2) >= (FP_MAX_SIZE / 8)) {
-            cli_errmsg("cli_loadcrt: line %u: Public key too long.\nNOTE: If this is actually a valid key length, recompile with a larger FP_MAX_SIZE (currently %d).\n", (unsigned int)line, FP_MAX_SIZE);
-            ret = CL_EMALFDB;
+        if (!BN_set_word(ca.e, 65537))
```

Comment:
A comment to explain the 65537 number would be helpful, or else using a constant/macro with a meaningful name. 

It would also be nice to have an error message in case it fails.

Reply:
This is the RSA exponent I haven't changed it. I just optimized it a little since it was disguised earlier as

`const uint8_t exp[] = "\x01\x00\x01";`

and then parsed via fp_read_unsigned_bin().  Let me add a comment and an error message as asked.